modules/MarketAnalysis.py

```diff
@@ -148,3 +225,87 @@ def get_percentile(self, rates, lending_style, use_numpy=use_numpy):
             result = self.percentile(sorted(rates), lending_style / 100.0)
         result = truncate(result, 6)
         return result
+
+    def get_golden_cross_rate(self, cur, rates_df, short_period=150, long_period=1800):
+        # TODO These don't need to be rolling mean, simple mean on the last X will do (and will be faster)
+        short_rate = pd.rolling_mean(rates_df[1], window=short_period, min_periods=1).iloc[-1]
+        long_rate = pd.rolling_mean(rates_df[1], window=long_period, min_periods=1).iloc[-1]
+        # TODO remove the sys writes
+        if short_rate > long_rate:
+            sys.stdout.write("Short higher : ")
+            rate = short_rate if rates_df.iloc[-1][1] < short_rate else rates_df.iloc[-1][1]
+        else:
+            sys.stdout.write("Long  higher : ")
+            rate = long_rate
+        # TODO Needs config option
+        rate = rate * 1.05
```

Comment:
The * 1.05?

The assumption here is that if you have a short rate that is rising above the long rate then the market is in a period where it's rising. So here when we return the rate we're trying to catch the top of the spike. It's not the best way to do it, but it works. 

I'd initially thought making this configurable would be a good way to implement this. Now though after chatting to @rnevet I think we both agree that this module should feed into the gapbottom and gaptop (I actually want to be able to drive those AND cur_daily_min via config from the MA module, possibly even with different methods, though that's long term).

So if I do it with the top/bottom method this will probably change. Here I was trying to get the best value for just one loan. When I was developing this I wasn't using spread at all and I was getting really good results. Though it's clear you can get even better results using the spread if it's clever enough.

In the new top/bottom method I'm going to take the trend of the short term moving average and extrapolate the plot to work out where the trend looks like it's moving to. The example I gave was this:

```TIME - Short - Long
12:00 - 0.08 - 0.1
12:01 - 0.09 - 0.1
12:02 - 0.1 - 0.1
12:03 - 0.11 - 0.1
12:04 - 0.12 - 0.1
```
In that example, the gapbottom would be 0.12 and gaptop would be 0.14. So we'd be making the assumption that the current upward trend continues.


I hope you meant the * 1.05, otherwise I've just gone on a massive rant! :)